src/core_plugins/input_control_vis/public/input_control_request_handler.js

```diff
@@ -0,0 +1,175 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import _ from 'lodash';
+import { timefilter } from 'ui/timefilter';
+import { buildPhraseFilter } from 'ui/filter_manager/lib/phrase';
+
+const termsAgg = (field, size, direction) => {
+  if (size < 1) {
+    size = 1;
+  }
+  const terms = {
+    size: size,
+    order: {
+      _count: direction
+    }
+  };
+  if (field.scripted) {
+    terms.script = {
+      inline: field.script,
+      lang: field.lang
+    };
+    terms.valueType = field.type === 'number' ? 'float' : field.type;
+  } else {
+    terms.field = field.name;
+  }
+  return {
+    'termsAgg': {
+      'terms': terms
+    }
+  };
+};
+
+const minMaxAgg = (field) => {
+  const aggBody = {};
+  if (field.scripted) {
+    aggBody.script = {
+      inline: field.script,
+      lang: field.lang
+    };
+  } else {
+    aggBody.field = field.name;
+  }
+  return {
+    maxAgg: {
+      max: aggBody
+    },
+    minAgg: {
+      min: aggBody
+    }
+  };
+};
+
+const createAncestorFilters = (controls, indexPattern, control) => {
+  let filters = [];
+  const parentControl = controls.find(c => c.id = control.parent);
+  if (parentControl.value) {
+    filters.push(buildPhraseFilter(parentControl.fieldName, parentControl.value, indexPattern));
+  }
+  if (parentControl.parent) {
+    filters = [
+      ...filters,
+      ...createAncestorFilters(controls, indexPattern, parentControl)
+    ];
+  }
+  return filters;
+};
+
+export const inputControlRequestHandler = async (vis, { searchSource, timeRange }) => {
+  const controlParams = vis.params;
+
+  const initialSearchSourceState = {
+    timeout: '1s',
+    terminate_after: 100000
+  };
+
+  // for each input control (maybe we can batch the requests ?)
+  return await Promise.all(vis.params.controls.map(async control => {
+    const controlData = {
+      enable: true,
+      disableReason: '',
+    };
+
+    let indexPattern;
+    try {
+      indexPattern = await vis.API.indexPatterns.get(control.indexPattern);
+    } catch (err) {
+      controlData.enable = false;
+      return controlData;
+    }
+
+    let ancestorFilters = [];
+    if (control.parent) {
+      ancestorFilters = createAncestorFilters(vis.params.controls, indexPattern, control);
+    }
+
+    const field = indexPattern.fields.byName[control.fieldName];
+    const aggs = control.type === 'list' ?
+      termsAgg(field, _.get(control.options, 'size', 5), 'desc') :
+      minMaxAgg(field);
+
+    const requestSearchSource = searchSource.makeChild(initialSearchSourceState);
+
+    requestSearchSource.filter(() => {
+      const activeFilters = [...ancestorFilters];
+      if (controlParams.useTimeFilter) {
```

Comment:
Maybe this check is not even needed any more. The original design of input controls was that the aggregation request would only run once. From then on, the list would never update. This confused some people so `useTimefilter` was added to make the it behave more like a traditional visualization and refetch the list when updated. I think if the aggregation request is going to be run each time the request handler is called then `useTimefilter` is no longer needed. @Stacey-Gammon What are your thoughts?